src/hooks/useForm.ts

```diff
@@ -1,10 +1,12 @@
 import { useState } from 'react'
 import { useNavigate } from 'react-router-dom'
-import { SignFormProps, SigninResponse } from '../components/common/Sign/types'
+import { SigninResponse } from '../components/common/Sign/types'
 import { postSignin, postSignup } from '../apis/auth'
 import { useAuthContext } from '../AuthProvider'
 
-const useForm = ({ isSignUp }: SignFormProps) => {
+export type FormModeType = 'signin' | 'signup'
+
```

Comment:
```suggestion
export type FormModeType = FormType.SignIn | FormType.SignUp
export enum FormType {
  SignIn = 'signin',
  SignUp = 'signup',
}
```

혹시 이렇게 enum을 따로 정의해서 사용해보는 것은 어떨까요?

타입이 정해져 있어 실수를 줄일 수 있어서 좋을 것 같은데, 어떻게 생각하세요?

Reply:
enum을 많이 사용해보지 않았는데, 좋은 방법인 듯 보여요!

Reply:
https://techblog.woowahan.com/9804

enum을 사용했을 때의 장단점이 잘 정리되어 있어서 가져와봤습니다!

js로 변환시 용량이 커진다고는 하는데, 그게 좀 걱정이네요 ㅎㅎ;